hypothesis_2.ipynb

```diff
@@ -3543,12 +3559,23 @@
     "\n",
     "        # GET request\n",
     "        response = session.get(url)\n",
+    "        # AW: minor thing, reversing the condition would reduce indendation.\n",
+    "        # Also, the \"!= 200\" branch should raise an exception or record an\n",
+    "        # error.\n",
     "        if response.status_code == 200:\n",
     "            # See request JSON\n",
     "            res = response.json()['query']['users']\n",
     "            print(res)\n",
     "\n",
     "            # save results to dictionary to process later\n",
+    "\n",
+    "            # AW: Although I like the encapsulation of each user's fetched\n",
```

Comment:
@adamwight 

Could you clarify?

Reply:
Might be easier to explain in a call.  Currently, the code is leaning towards a design where data about each user is built up in a subdirectory for each user.  This would make sense if we were using a database store, but on the filesystem this will result in a really huge number of files (hundreds of thousands of users * dozens or hundreds of wikis) which is tough for system performance and very awkward to eg. push to network storage.

A bigger issue is that it doesn't fit the logical flow of fetching the data, which is first of all partitioned by language.  So if our process gets interrupted then we have partial information about all users for some languages, and we just have to start over again.  By contrast, if we were organizing the output data by language then an interrupted program can restart by skipping over all completed languages.